packages/slo-stat-bot/src/slo-compliant.ts

```diff
@@ -0,0 +1,377 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// eslint-disable-next-line node/no-extraneous-import
+import {GitHubAPI} from 'probot';
+import moment from 'moment';
+import {SLORules, IssuesListCommentsItem} from './types';
+import {convertToArray} from './slo-appliesTo';
+
+interface IssueAssignees {
+  login: string;
+}
+
+interface ReposListCollaboratorsItem {
+  login: string;
+  permissions: {
+    pull: boolean;
+    push: boolean;
+    admin: boolean;
+  };
+}
+
+/**
+ * Function gets the file contents
+ * @param github unique installation id for each function
+ * @param owner of issue or pr
+ * @param repo of issue or pr
+ * @param path of the file
+ * @returns string of the content in the file
+ */
+export const getFilePathContent = async function getFilePathContent(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  path: string
+): Promise<string> {
+  try {
+    const fileResponse = await github.repos.getContents({
+      owner,
+      repo,
+      path,
+    });
+    const data = fileResponse.data as {content?: string};
+    const content = Buffer.from(data.content as string, 'base64').toString(
+      'utf8'
+    );
+    return content;
+  } catch (err) {
+    return 'not found';
```

Comment:
I think we want to start throwing errors if we don't recognize them... see #722. See if you can catch the normal errors you'd get, and maybe let's use the body of the error message to surface a usable error. For other errors, we can just let it throw if that goes along with the execution of the bot (i.e., if you ultimately want the bot to fail there anyways).